src/NServiceBus.Core/BusConfiguration.cs

```diff
@@ -267,11 +267,18 @@ internal InitializableEndpoint Build()
         {
             ForAllTypes<T>(types, t =>
             {
+                if (!HasDefaultConstructor(t))
+                {
+                    throw new Exception($"Unable to create the type '{t.Name}'. Types implementing '{typeof(T).Name}' must have a public parameterless (default) constructor.");
+                }
+
                 var instanceToInvoke = (T)Activator.CreateInstance(t);
                 action(instanceToInvoke);
             });
         }
 
+        static bool HasDefaultConstructor(Type type) => type.GetConstructor(Type.EmptyTypes) != null;
```

Comment:
if we already reflect to get the default constructor why not use that to create the instance